pkg/node/manager/events.go

```diff
@@ -0,0 +1,119 @@
+package manager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/benbjohnson/clock"
+
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+)
+
+// NodeEvent represents the type of event that can be emitted by the NodeEventEmitter.
+type NodeEvent int
+
+const (
+	NodeEventApproved NodeEvent = iota
+	NodeEventRejected
+	NodeEventDeleted
+	NodeEventConnected
+	NodeEventDisconnected
+)
+
+func (n NodeEvent) String() string {
+	if n == NodeEventApproved {
+		return "NodeEventApproved"
+	} else if n == NodeEventRejected {
+		return "NodeEventRejected"
+	} else if n == NodeEventDeleted {
+		return "NodeEventDeleted"
+	} else if n == NodeEventConnected {
+		return "NodeEventConnected"
+	} else if n == NodeEventDisconnected {
+		return "NodeEventDisconnected"
+	}
+
+	return "UnknownNodeEvent"
+}
+
+type NodeEventEmitterOption func(emitter *NodeEventEmitter)
+
+// WithClock is an option that can be used to set the clock for the NodeEventEmitter. This is useful
+// for testing purposes.
+func WithClock(clock clock.Clock) NodeEventEmitterOption {
+	return func(emitter *NodeEventEmitter) {
+		emitter.clock = clock
+	}
+}
```

Comment:
Not sure this is used anywhere, was it meant to be used in a test?

Reply:
It was, and is with b9a69535f